### PR TITLE
[PICK] Add new common version file for projects that have release streams outside of OS and EE

### PR DESCRIPTION
**Pull request by Brian-McM**
## Description

Pick of https://github.com/tigera/operator/pull/1150

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: config/common_versions.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+title: master
```
Comment (caseydavenport) [inline config/common_versions.yml:1]: Is master correct for this branch?

Comment (Brian-McM) [reply config/common_versions.yml]: Probably not... I updated it, I think that's correct?

File: config/common_versions.yml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+title: v3.17.1
```
Comment (tmjd) [inline config/common_versions.yml:1]: Is title needed here? I can imagine it is. If required I'd prefer it be set to unused or something like that.

File: config/common_versions.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+components:
+  key-cert-provisioner:
+    image: tigera/key-cert-provisioner
+    version: v1.0.0
```
Comment (tmjd) [inline config/common_versions.yml:5]: Does this image exist already? I would have though this should be set to release-v1.0

File: config/common_versions.yml (hunk 2 above)
Comment (Brian-McM) [reply config/common_versions.yml:1]: So literally put in "unused"?

File: config/common_versions.yml (hunk 3 above)
Comment (Brian-McM) [reply config/common_versions.yml:5]: I was just following the conventions of the other versions, I'll change it to release-v1.0.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**